src/preact/base-element.js

```diff
@@ -725,6 +728,21 @@ function collectProps(Ctor, element, ref, defaultProps) {
   return props;
 }
 
+/**
+ * Clones an Element into a VNode.
+ * @param {!Element} element
+ * @return {!PreactDef.Renderable}
+ */
+function shallowCloneVNode(element) {
+  const {attributes} = element;
+  const props = {key: element};
+  for (let i = 0; i < attributes.length; i++) {
```

Comment:
Please cache the `length` in a variable. `attributes` is a live `NamedNodeMap`, and the regular `array.length` loop optimizations do not apply to it.

Reply:
TIL, done.